mods/ra/maps/soviet-03/soviet03.lua

```diff
@@ -0,0 +1,387 @@
+	paraproxy = Actor.Create("powerproxy.paratroopers", false, { Owner = player })
+	paraproxy.SendParatroopers(ReinforcementDropOff.CenterPosition, false, 0)
+	paraproxy.Destroy()
+end
+
+SpyFinalSequency = function()
+	if not SpyHideout4.IsDead then
+		SpyHideout4.UnloadPassengers()
+		Trigger.AfterDelay(DateTime.Seconds(1), function()
+			if not TheSpy.IsDead then
+				TheSpy.Move(SpyGoal.Location)
+			end
+		end)
+	end
+end
+
+SpyHelicopterEscape = function()
+	if not spyHelicopterEscape then
+		spyHelicopterEscape = true
+		SpyFinalSequency()
+		Actor.Create("camera", true, { Owner = player, Location = CameraFinalArea.Location })
+		ExtractionHeli = Reinforcements.ReinforceWithTransport(germany, ExtractionHeliType, nil, ExtractionPath)[1]
+		local exitPos = CPos.New(ExtractionPath[1].X, ExtractionPath[2].Y)
+		Trigger.AfterDelay(DateTime.Seconds(5), function()
+			if not TheSpy.IsDead and not ExtractionHeli.IsDead then
+				TheSpy.EnterTransport(ExtractionHeli)
+			end
+		end)
+		Trigger.AfterDelay(DateTime.Seconds(7), function()
+			if not ExtractionHeli.IsDead then
+				ExtractionHeli.Move(HelicopterEscape.Location)
+			end
+		end)
+		Trigger.AfterDelay(DateTime.Seconds(12), function()
+			enemy.MarkCompletedObjective(alliedObjective)
+		end)
+	end
+end
+
+Trigger.OnAllKilled(Farmers, function()
+	Reinforcements.Reinforce(player, USSRReinforcementsFarm, { FarmSpawn.Location, FarmArea.Location }, 0)
+	player.MarkCompletedObjective(sovietObjective2)
+end)
+
+Trigger.OnAllKilled(RedBuildings, function()
+	player.MarkCompletedObjective(sovietObjective3)
+end)
+
+Trigger.OnAnyKilled(BarrierSoldiers, function()
+	if barrier1Trigger then
+		Utils.Do(BarrierSoldiers, function(actor)
+			if not actor.IsDead then
+				Trigger.OnIdle(actor, actor.Hunt)
+			end
+		end)
+	end
+end)
+
+Trigger.OnEnteredFootprint(SpyHideout1Trigger, function(a, id)
+	if not spyHideout1Trigger and a.Owner == player then
+		spyHideout1Trigger = true
+		Trigger.RemoveFootprintTrigger(id)
+		Actor.Create("camera", true, { Owner = player, Location = SpyHideout1.Location })
+		if not TheSpy.IsDead and SpyHideout1.IsDead then
+			TheSpy.EnterTransport(SpyHideout1)
+		end
+	end
+end)
+
+Trigger.OnEnteredFootprint(SpyHideout2PathTrigger, function(a, id)
+	if not spyHideout2PathTrigger and a.Owner == player then
+		spyHideout2PathTrigger = true
+		Trigger.RemoveFootprintTrigger(id)
+		Actor.Create("camera", true, { Owner = player, Location = CameraSpyVillage.Location })
+		Actor.Create("camera", true, { Owner = player, Location = CameraVillage.Location })
+		if not TheSpy.IsDead and not SpyHideout2.IsDead then
+			TheSpy.EnterTransport(SpyHideout2)
+		end
+	end
+end)
+
+Trigger.OnEnteredFootprint(SpyHideout2Trigger, function(a, id)
+	if not spyHideout2Trigger and a.Owner == player then
+		spyHideout2Trigger = true
+		SpyGuards1 = Reinforcements.Reinforce(germany, EnemyReinforcements1SpyHideout2, { EnemyReinforcements1.Location, EnemyReinforcements1Goal.Location }, 0)
```

Comment:
We either need a check here if the house in which those guys are hiding is still there, or you expand the Footprint, as you can snipe the house with grenadiers before triggering the cell trigger.